### DISCUSSION TITLE
I have built c:geo myself :)

### DISCUSSION THREAD

**Discussion by Lineflyer** [cgeo/cgeo#10184] [Show and tell]
After many years of somehow contributing here, I finally found some time to try to setup a build environment for the very first time.
It was mainly an easy task by installing Android Studio and importing the project using the readme.md and further Wiki pages.

One problem I had:
When trying to build via Gradle it always tried to use a Java version defined by the environment variable `JAVA_HOME` instead of using the JDK8 which Android Studio has installed. Looking into File - Project Structure - SDK Location its clearly defined, that Gradle shall use the Java SDK in then Android studio path `C:\Program Files\Android\Android Studio\jre`. The gradle build is however ignoring this, thus the build failed as the environment variable just targets to some runtime environment (or even an incomplete install).
I wonder whether any of you has a solution other than redirecting the environment variable to the Anroid Studio path.

Anyway finally I was able to build the project and use c:geo in the emulator... Hurraay!

Now my vacation is over and there is less time to work on that.
However, some questions for best practice for the next time I continue to fine tune that:
- Do you control git from within Android Studio or just plain on CLI or even with just another tool?
- How do I trigger a real new build and new installation in Android Studio. Right now it seems it buffers build and emulator (takes few seconds to build and emulator always comes up in last status)? Reason: I provided a GMaps key in `private.properties` and would like to test if it works...but it just does not build again.
-

**Comment by moving-bits** [comment]
> I wonder whether any of you has a solution other than redirecting the environment variable to the Anroid Studio path.

Hm, don't remember - setting up my environment is more than two years ago... But checking the command line I can see that my `JAVA_HOME` is set to Android Studio.

Ah, just cross-checked: I'm heavily using a couple of batch files for switching projects, changing to my dev dir, updating my local repo etc. The main script also includes setting `JAVA_HOME`.

> Anyway finally I was able to build the project and use c:geo in the emulator... Hurraay!

Congrats & welcome to the world of devs. Expecting regular code contributions from now on ;-))

> Do you control git from within Android Studio or just plain on CLI or even with just another tool?

Switching repo and updating from server I do with some batch files. Checking changes for a specific code block (aka "blaming") I do from within AS. Everything else I do using the command line.

> How do I trigger a real new build

You can use `File` => `Invalidate caches` to force AS to rebuild everything from scratch. I do need it only in some corner cases when AS has sort of a "hiccup". But I guess it does NOT refill the template files - would need to check. (But you could use a script for it.)

> and new installation in Android Studio. Right now it seems it buffers build and emulator (takes few seconds to build and emulator always comes up in last status)? Reason: I provided a GMaps key in `private.properties` and would like to test if it works...but it just does not build again.

To force AS to rebuild your `keys.xml` you can simply delete the existing one (either manually or triggered by a script).

Regarding the emulator: You can force a cold boot. And you can uninstall and reinstall c:geo, which provides you with a clean environment (at least regarding the internal folders).

**Comment by fm-sys** [comment]
Cool! I have something to share as well. Today I've generated my first Google Maps API key  :)

It was possible for me without entering a credit card number... Here are two links in case someone wants to try it:
1. Sign in to the developer console: https://console.developers.google.com/
2. Generate API key: https://console.cloud.google.com/flows/enableapi?apiid=maps_android_backend&keyType=CLIENT_SIDE_ANDROID
